docs/core/plugins/sieve.md

```diff
@@ -46,128 +76,207 @@ protocol lmtp {
+
+The `default` storage yields the sieve script that gets executed **only** if
+the user's personal Sieve script does not exist. Although more
+than a single `default` storage can be defined, only the first one listed in
+the configuration is used. 
+
+If [[setting,sieve_script_name]] is set for this script storage, the default
+script can be seen and accessed by this name through ManageSieve (and
+doveadm sieve). See below ([[link,sieve_visible_default_script]]).
+
+This is usually a global script, so be sure to pre-compile the specified
+script manually in that case using the sievec command line tool, as
+explained by [[man,sievec]].`
+
+#### `discard` {#script-storage-type-discard}
+
+The `discard` storage yields the sieve script that gets executed for any message
+that is about to be discarded; i.e., it is not delivered anywhere by the normal
+Sieve execution. Although more than a single `discard` storage can be defined,
+only the first one listed in the configuration is used. The `discard` storage
+is currently only applicable for message delivery.
+
+The script from the `discard` storage is only executed when the "implicit keep"
+is canceled, by e.g. the "discard" action, and no actions that deliver the
+message are executed. Delivery in this case means both local delivery to a
+mailbox and redirection to a remote recipient. This "discard script" can prevent
+discarding the message, by executing alternative actions. If the discard
+script does nothing, the message is still discarded as it would be when no
+discard script is configured.
+
+#### `global` {#script-storage-type-global}
+
+An `global` storage is the source of `:global` include scripts for the Sieve
+include extension. Scripts are accessed by name, so if the storage yields only
+one script, a name must be defined for it; either implicitly by the storage
+driver or explicitly using [[setting,sieve_script_name]]. Multiple `global`
+storages can be configured and each storage will be queried in sequence to
+retrieve the requested script by name. The storages will be queried in the order
+these storages are defined in the configuration until the script is found. The
+order can be overridden by the [[setting,sieve_script_precedence]] setting.
+
+### Script storage drivers
+
+Sieve script storages are implemented as a storage driver. The default
+[file](#sieve_storage_file) storage driver uses the local filesystem. It can use
+a single script file or a directory containing several Sieve script files with a
+symbolic link pointing to the active script.
+
+More complex setups can use other storage drivers such as
+[ldap](#sieve_storage_ldap) or [dict](#sieve_storage_dict) to fetch Sieve
+scripts from LDAP databases or dict storages, respectively.
+
+The storage driver is configured using the [[setting,sieve_script_driver]]
+setting. If not explicitly configured for a [[setting,sieve_script]] block, the
+storage driver is `file` and the default directory is `~/sieve/` with a symbolic
```

Comment:
`is [[link,sieve_storage_file,file]]` and`

Reply:
fixed